library/network_connections.py

```diff
@@ -1559,24 +1559,36 @@ def _complete_kwargs_loglines(self, rr, connections, idx):
                 c["persistent_state"],
             )
             prefix = prefix + (", '%s'" % (c["name"]))
-        for r in rr["log"]:
-            yield (r[2], "[%03d] %s %s: %s" % (r[2], LogLevel.fmt(r[0]), prefix, r[1]))
+        for severity, msg, idx in rr["log"]:
+            yield (
+                idx,
+                "[%03d] %s %s: %s" % (idx, LogLevel.fmt(severity), prefix, msg),
+                severity,
+            )
 
-    def _complete_kwargs(self, connections, kwargs, traceback_msg=None):
+    def _complete_kwargs(self, connections, kwargs, traceback_msg=None, fail=False):
         if "warnings" in kwargs:
```

Comment:
You  can remove the full if/else block with just

```python
warning_logs = kwargs.get("warnings", [])
```

It works regardless of whether "warnings" is in kwargs.

Reply:
You are right, I didn't see that, thanks!